dialogflow/detect_intent_audio.rb

```diff
@@ -0,0 +1,78 @@
+# Copyright 2018 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "securerandom"
+
+def detect_intent_audio project_id:, session_id:, audio_file_path:,
+                        language_code:
+  # [START dialogflow_detect_intent_audio]
+  # project_id = "Your Google Cloud project ID"
+  # session_id = "mysession"
+  # language_code = "en-US"
+  
+  require "google/cloud/dialogflow"
+
+  session_client = Google::Cloud::Dialogflow::Sessions.new
+  session = session_client.class.session_path project_id, session_id
+  puts "Session path: #{session}"
+
+  begin
+    audio_file = File.open(audio_file_path, "rb")
+    input_audio = audio_file.read
+  ensure
+    audio_file.close
+  end
+
+  audio_config = { 
+    audio_encoding: :AUDIO_ENCODING_LINEAR_16,
+    sample_rate_hertz: 16000,
+    language_code: language_code
+  }
+
+  query_input = { audio_config: audio_config }
+
+  response = session_client.detect_intent session, query_input, input_audio: input_audio
+  query_result = response.query_result
+
+  puts "=" * 20
```

Comment:
Let's not use this cute, fancy Ruby feature to add separators ;)

It took me a moment to understand what this is doing, at first glance it's confusing. The sample is complex enough, let's not do this :)

We're only printing out 4 lines per Intent, so I wouldn't add a separator at all for this one

I tend to stay away from anything in the sample code that exists only to format the CLI output, eg. I also prefer to not see `\t` and `\n` because it's irrelevant to the code & just adds extra syntax fluff

Reply:
For this one, in particular, `"=" * 20` is too _**cleaver**_ :P

Reply:
done.